core/sup_tree_core/periodic_log/message_builder.ex

```diff
@@ -0,0 +1,38 @@
+# Copyright(c) 2015-2019 ACCESS CO., LTD. All rights reserved.
+
+defmodule AntikytheraCore.PeriodicLog.MessageBuilder do
+  @max_proc_to_log     5
+  @max_msg_to_log      10
+  @queue_len_threshold 100
+
+  def init() do
+    nil
+  end
+
+  def build_log(state) do
+    procs =
+      :recon.proc_count(:message_queue_len, @max_proc_to_log)
+      |> Enum.filter(fn({_pid, qlen, _info}) -> qlen >= @queue_len_threshold end)
+    if procs != [] do
```

Comment:
Since `if` returns `nil` when `procs == []`, I think the following can reduce code duplication:

```elixir
    log =
      if procs != [] do
        log_time = Antikythera.Time.to_iso_timestamp(Antikythera.Time.now())
        procs
        #...
      end
    {log, state}
```

Reply:
Instead, I extracted into separate functions. 0a6e8eef1604543b6f9e22b1d92b945aa1b50bdb